ui_advanced_config.py

```diff
@@ -617,291 +628,511 @@ def updateComponentsEntryWidgets(self, v):
-                    self.componentTypeAttr[9]
-                ),
-            )
-        elif (
-            self.componentData[self.componentIDs[currentComponentIdx]].getData("ctype")
-            == "Radio"
-        ):
+                0, currentComp.getData(self.componentTypeAttr[9])
+            )
+        elif currentComp.getData("ctype") == "Radio":
             self.componentsTypeAttr1Label.config(text=self.componentTypeAttr[4])
             self.componentsTypeAttr1Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[4]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[4])
             )
             self.componentsTypeAttr2Label.config(text=self.componentTypeAttr[5])
             self.componentsTypeAttr2Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[5]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[5])
             )
             self.componentsTypeAttr3Label.config(text=self.componentTypeAttr[6])
             self.componentsTypeAttr3Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[6]
-                ),
-            )
-        elif (
-            self.componentData[self.componentIDs[currentComponentIdx]].getData("ctype")
-            == "Arm"
-        ):
+                0, currentComp.getData(self.componentTypeAttr[6])
+            )
+        elif self.currentComponentData.getData("ctype") == "Arm":
             self.componentsTypeAttr1Label.config(text=self.componentTypeAttr[4])
             self.componentsTypeAttr1Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[4]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[4])
             )
             self.componentsTypeAttr2Label.config(text=self.componentTypeAttr[5])
             self.componentsTypeAttr2Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[5]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[5])
             )
             self.componentsTypeAttr3Label.config(text=self.componentTypeAttr[6])
             self.componentsTypeAttr3Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[6]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[6])
             )
 
-    def updateObjectsEntryWidgets(self, v):
```

Comment:
Similar problems to teams and components (can create without ID or with only ID --> cause problems), but this time I noticed that I can do it twice and have two objects with the same ID.
<img width="323" alt="Screen Shot 2024-03-02 at 1 04 05 AM" src="https://github.com/Leading-AI-Development-Solutions/MAIA/assets/100381732/5aae9380-5d98-4f97-9c0c-707dcef7d26c">

Reply:
Better validation should fix this I think?